server/routes/api/disease.py

```diff
@@ -32,3 +35,29 @@ def get_node(dcid):
                                    post=False,
                                    has_payload=False)
   return response
+
+
+@bp.route('/diseaseParent/<path:dcid>')
+def get_disease_parent(dcid):
+  """Returns parent node for given a disease node."""
+  list_dcid = []  ## list to store disease dcids
+  list_name = []  ## list to store disease names
+  final_dcid = dcid
+  while (final_dcid != "bio/DOID_4"
+        ):  ## dcid of the biggest parent node where iteration stops
+    node_dcids = dc_service.property_values([final_dcid],
+                                            "specializationOf").get(
+                                                final_dcid, [])
+    if not node_dcids:
+      break
+    node_dcid = node_dcids[0]
+    node_names = dc_service.property_values([node_dcid],
+                                            "name").get(node_dcid, [])
+    node_name = node_dcid
+    if node_names:
+      node_name = node_names[0]
+    list_dcid.append(node_dcid)
+    list_name.append(node_name)
+    final_dcid = node_dcid
+  result = [list_dcid, list_name]  ## return a list of dcid and name lists
```

Comment:
Good idea! I first declared a class which stores the object of type `DiseaseParent`. Since the object was not JSON Serializable, I declared a sub class called `DiseaseParentEncoder` which solves the issue. Here's an example of the returned output from the json dump of DiseaseParent class - 
[{"dcid": "bio/DOID_0060036", "name": "intrinsic cardiomyopathy"}, {"dcid": "bio/DOID_0050700", "name": "cardiomyopathy"}, {"dcid": "bio/DOID_114", "name": "heart disease"}, {"dcid": "bio/DOID_1287", "name": "cardiovascular system disease"}, {"dcid": "bio/DOID_7", "name": "disease of anatomical entity"}, {"dcid": "bio/DOID_4", "name": "disease"}]